polaris-ui/src/app/features/cases/presentation/case-details/KeyDetails.tsx

```diff
@@ -18,20 +30,48 @@ export const KeyDetails: React.FC<{ caseDetails: CaseDetails }> = ({
         {caseDetails.uniqueReferenceNumber}
       </h1>
 
-      <div className={`govuk-heading-s`} data-testid="txt-defendant-name">
-        <div>
-          {caseDetails.leadDefendantDetails.surname},{" "}
-          {caseDetails.leadDefendantDetails.firstNames}
-        </div>
-        <div className={`${classes.namesub}`}>
-          DOB:
-          {formatDate(
-            caseDetails.leadDefendantDetails.dob,
-            CommonDateTimeFormats.ShortDateTextMonth
-          )}
-          . Age: {getAgeFromIsoDate(caseDetails.leadDefendantDetails.dob)}
+      {isMultipleDefendantsOrCharges && (
+        <>
+          <ul
+            className={classes.defendantsList}
+            data-testid="list-defendant-names"
+          >
+            {defendantsList.map(({ defendantDetails }) => (
+              <li key={defendantDetails.id}>
+                {defendantDetails.surname}, {defendantDetails.firstNames}
+              </li>
+            ))}
+          </ul>
+          <LinkButton
+            data-testid="link-defendant-details"
+            className={classes.defendantDetailsLink}
+            onClick={() => {}}
+          >
+            {`View ${defendantsList.length} ${
+              defendantsList.length > 1 ? "defendants" : "defendant"
+            } and charges`}
+          </LinkButton>
+        </>
+      )}
+      {!isMultipleDefendantsOrCharges && (
+        <div
+          className={classes.defendantDetails}
+          data-testid="defendant-details"
+        >
+          <span className={`govuk-heading-s ${classes.defendantName}`}>
+            {defendantsList[0].defendantDetails.surname},{" "}
```

Comment:
On the search results page, we use `leadDefendant`.  I wonder if we should use that here too? Once you have sorted the defendants list,  if the top defendant was different to `leadDefendant` then it would be confusing for the user?

Reply:
This  will happen only in the case of single defendant. For multiple defendants, we wont be ending up here.

Reply:
i have reverted  these changes , as there can be a scenario where defendant can be empty array even though we have leadDefendant details